### PR TITLE
Labels for Range Rings created outside of Basemap extent

### PR DESCRIPTION
**Pull request by kgonzago**
This fix addresses #533 for range rings. Logic implemented:
1. Gets the basemap extents
2. Creates a non-geodesic circle
3. Checks if circle crosses beyond basemap extent
 a. If so, then create label based on non-geodesic circle
 b. If not, then create label based on geodesic range ring circle

@csmoore Can you review? Thanks.

### CODE REVIEW

File: source/addins/ArcMapAddinDistanceAndDirection/ArcMapAddinDistanceAndDirection/ViewModels/TabBaseViewModel.cs [hunk 1]
```diff
@@ -941,6 +941,68 @@ internal ILayer GetLayerFromMapByName(IMap map, string name)
...
+        internal IEnvelope GetBasemapExtent(IMap map)
+        {
+            try
+            {
+                ILayer basemapLayer = null;
+                var layers = map.get_Layers();
+                var layer = layers.Next();
+                while (layer != null)
+                {
+                    if (layer is ICompositeLayer)
```
Comment (csmoore) [inline source/addins/ArcMapAddinDistanceAndDirection/ArcMapAddinDistanceAndDirection/ViewModels/TabBaseViewModel.cs]: I'm not sure this is a full-proof way of detecting the BaseMapLayer:

1. many different types of layers implement this interface (so just the first group layer would be returned)
2. user may be using just a simple layer/feature class as basemap (ex. world/continents shapefile)

Would it be possible just to use ActiveView.FullExtent (I think that is what is used when you hit the zoom to full extent button)?

Comment (kgonzago) [reply source/addins/ArcMapAddinDistanceAndDirection/ArcMapAddinDistanceAndDirection/ViewModels/TabBaseViewModel.cs]: @csmoore I'll test with full extent

**Comment by kgonzago** [comment]
@csmoore `IActiveView::FullExtent()` didn't work with the logic. So, I added `IBasemapLayer` for layer type check in the `GetBasemapExtent` function.

**Comment by kgonzago** [comment]
Added logic to handle for situation where a map might not have a basemap or multiple basemaps.

**Pull request merged**